alws/utils/pulp_client.py

```diff
@@ -67,6 +77,80 @@ async def get_rpm_remote(self, name: str) -> typing.Union[dict, None]:
             return None
         return response['results'][0]
 
+    async def create_module(self, content: str):
+        ENDPOINT = 'pulp/api/v3/content/rpm/modulemds/'
+        artifact_href, sha256 = await self.upload_file(content)
+        module = ModuleWrapper.from_template(content)
+        payload = {
+            'relative_path': 'modules.yaml',
+            'artifact': artifact_href,
+            'name': module.name,
+            'stream': module.stream,
+            # Instead of real module version, we're inserting
+            # mocked one, so we can update template in the future,
+            # since pulp have this global index:
+            # unique_together = ("name", "stream", "version", "context",
+            #                    "arch")
+            'version': get_random_unique_version(),
+            'context': module.context,
+            'arch': module.arch,
+            'artifacts': [],
+            'dependencies': []
+        }
+        task = await self.make_post_request(ENDPOINT, data=payload)
+        task_result = await self.wait_for_task(task['task'])
+        return task_result['created_resources'][0], sha256
+
+    async def check_if_artifact_exists(self, sha256: str) -> str:
+        ENDPOINT = 'pulp/api/v3/artifacts/'
+        payload = {
+            'sha256': sha256
+        }
+        response = await self.make_get_request(ENDPOINT, params=payload)
+        if response['count']:
+            return response['results'][0]['pulp_href']
+
+    async def _upload_file(self, content, sha256):
```

Comment:
What will be if file is big? In pulp uploader in build node we split file in parts and then send them